cli/src/main/java/org/apache/falcon/cli/commands/FalconEntityCommands.java

```diff
@@ -18,9 +18,304 @@
+import static org.apache.falcon.cli.FalconEntityCLI.SUMMARY_OPT_DESCRIPTION;
+import static org.apache.falcon.cli.FalconEntityCLI.SUSPEND_OPT;
+import static org.apache.falcon.cli.FalconEntityCLI.SUSPEND_OPT_DESCRIPTION;
+import static org.apache.falcon.cli.FalconEntityCLI.TAGKEYS_OPT;
+import static org.apache.falcon.cli.FalconEntityCLI.TAGKEYS_OPT_DESCRIPTION;
+import static org.apache.falcon.cli.FalconEntityCLI.TAGS_OPT;
+import static org.apache.falcon.cli.FalconEntityCLI.TAGS_OPT_DESCRIPTION;
+import static org.apache.falcon.cli.FalconEntityCLI.TYPE_OPT;
+import static org.apache.falcon.cli.FalconEntityCLI.TYPE_OPT_DESCRIPTION;
+import static org.apache.falcon.cli.FalconEntityCLI.UPDATE_OPT;
+import static org.apache.falcon.cli.FalconEntityCLI.UPDATE_OPT_DESCRIPTION;
+import static org.apache.falcon.cli.FalconEntityCLI.VALIDATE_OPT;
+import static org.apache.falcon.cli.FalconEntityCLI.VALIDATE_OPT_DESCRIPTION;
+import static org.apache.falcon.cli.FalconEntityCLI.validateEntityFields;
+
 /**
  * Entity Commands.
  */
+@Component
 public class FalconEntityCommands extends BaseFalconCommands {
+    public static final String ENTITY_PREFIX = "entity";
+    public static final String ENTITY_COMMAND_PREFIX = ENTITY_PREFIX + " ";
+
+    @CliCommand(value = ENTITY_COMMAND_PREFIX + SLA_MISS_ALERT_OPT, help = SLA_MISS_ALERT_OPT_DESCRIPTION)
+    public String slaAlert(
+            @CliOption(key = {TYPE_OPT}, mandatory = true, help = TYPE_OPT_DESCRIPTION) final EntityType entityType,
+            @CliOption(key = {ENTITY_NAME_OPT}, mandatory = true, help = ENTITY_NAME_OPT_DESCRIPTION)
+            final String entityName,
+            @CliOption(key = {COLO_OPT}, mandatory = true, help = COLO_OPT_DESCRIPTION) final String colo,
+            @CliOption(key = {START_OPT}, mandatory = false, help = START_OPT_DESCRIPTION) final String start,
+            @CliOption(key = {END_OPT}, mandatory = false, help = END_OPT_DESCRIPTION) final String end
+    ) {
+        SchedulableEntityInstanceResult response = getFalconClient()
+                .getFeedSlaMissPendingAlerts(entityType.name().toLowerCase(), entityName, start, end, getColo(colo));
+        return ResponseHelper.getString(response);
+    }
+
+    @CliCommand(value = ENTITY_COMMAND_PREFIX + SUBMIT_OPT, help = SUBMIT_OPT_DESCRIPTION)
+    public String submit(
+            @CliOption(key = {TYPE_OPT}, mandatory = true, help = TYPE_OPT_DESCRIPTION) final EntityType entityType,
+            @CliOption(key = {FILE_PATH_OPT}, mandatory = true, help = FILE_PATH_OPT_DESCRIPTION) final File filePath
+    ) {
+
+        return getFalconClient().submit(entityType.name().toLowerCase(), filePath.getPath(), getDoAs()).getMessage();
+    }
+
+    @CliCommand(value = ENTITY_COMMAND_PREFIX + LOOKUP_OPT, help = LOOKUP_OPT_DESCRIPTION)
+    public String lookup(
+            @CliOption(key = {TYPE_OPT}, mandatory = true, help = TYPE_OPT_DESCRIPTION) final EntityType entityType,
```

Comment:
I believe such a validation should be done in server. That way the validation will be centralized. Otherwise all clients need to add validation code themselves. Given the clients can be in java/python/node/ruby/..., each needs to implement validations.